api/src/Security/Authenticator/EventListener/UserLoaderListener.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Security\Authenticator\EventListener;
+
+use App\Security\Authenticator\Passport\Badge\TokenBadge;
+use App\Security\Core\TokenAwareUserProviderInterface;
+use Lcobucci\JWT\Token;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+use Symfony\Component\Security\Http\Event\CheckPassportEvent;
+
+/**
+ * @author Vincent Chalamon <vincentchalamon@gmail.com>
+ */
+final class UserLoaderListener
+{
+    public function __construct(private UserProviderInterface $userProvider)
+    {
+    }
+
+    public function checkPassport(CheckPassportEvent $event): void
+    {
+        $passport = $event->getPassport();
+        if (!$passport->hasBadge(TokenBadge::class)) {
+            return;
+        }
+
+        /** @var TokenBadge $badge */
+        $badge = $passport->getBadge(TokenBadge::class);
+        if (null !== $badge->getUserLoader()) {
+            return;
+        }
+
+        if (!$this->userProvider instanceof TokenAwareUserProviderInterface) {
+            return;
+        }
+
+        if (method_exists($this->userProvider, 'loadUserByIdentifier')) {
+            $badge->setUserLoader(function (string $identifier, ?Token $token): UserInterface {
+                /* @phpstan-ignore-next-line */
+                return $this->userProvider->loadUserByIdentifier($identifier, $token);
+            });
+
+            return;
+        }
+
+        trigger_deprecation('symfony/security-http', '5.3', 'Not implementing method "loadUserByIdentifier()" in user provider "%s" is deprecated. This method will replace "loadUserByUsername()" in Symfony 6.0.', get_debug_type($this->userProvider));
```

Comment:
do you need these in your userland code?
or this is because some of these file are intended to end up in a bundle or in core?

Reply:
This file intends to end up in symfony/security component

Reply:
If it ends up in the security component, it will be on 6+ where `loadUserIdentifier()` is mandatory anyway so it's safe to remove this notice